pkgs/applications/graphics/sane/backends/default.nix

```
@@ -1,57 +1,89 @@
-, gt68xxFirmware ? null, snapscanFirmware ? null
+  gt68xxFirmware ? null, snapscanFirmware ? null,
 
 # Not included by default, scan snap drivers require fetching of unfree binaries.
-, scanSnapDriversUnfree ? false, scanSnapDriversPackage ? sane-drivers.epjitsu
+  scanSnapDriversUnfree ? false, scanSnapDriversPackage ? sane-drivers.epjitsu,
 }:
 
-stdenv.mkDerivation {
+stdenv.mkDerivation rec {
   pname = "sane-backends";
-  version = "1.2.1";
+  version = "1.3.1";
 
-  src = fetchurl {
-    # raw checkouts of the repo do not work because, the configure script is
-    # only functional in manually uploaded release tarballs.
-    # https://gitlab.com/sane-project/backends/-/issues/440
-    # unfortunately this make the url unpredictable on update, to find the link
-    # go to https://gitlab.com/sane-project/backends/-/releases and choose
-    # the link under the heading "Other".
-    url = "https://gitlab.com/sane-project/backends/uploads/110fc43336d0fb5e514f1fdc7360dd87/sane-backends-1.2.1.tar.gz";
-    sha256 = "f832395efcb90bb5ea8acd367a820c393dda7e0dd578b16f48928b8f5bdd0524";
+  src = fetchFromGitLab {
+    owner = "sane-project";
+    repo = "backends";
+    rev = "refs/tags/${version}";
+    hash = "sha256-4mwPGeRsyzngDxBQ8/48mK+VR9LYV6082xr8lTrUZrk=";
   };
 
-  patches = [
-    # sane-desc will be used in postInstall so compile it for build
-    # https://github.com/void-linux/void-packages/blob/master/srcpkgs/sane/patches/sane-desc-cross.patch
-    (fetchpatch {
-      name = "compile-sane-desc-for-build.patch";
-      url = "https://raw.githubusercontent.com/void-linux/void-packages/4b97cd2fb4ec38712544438c2491b6d7d5ab334a/srcpkgs/sane/patches/sane-desc-cross.patch";
-      sha256 = "sha256-y6BOXnOJBSTqvRp6LwAucqaqv+OLLyhCS/tXfLpnAPI=";
-    })
-  ];
-
   postPatch = ''
+    # Do not create lock dir in install phase
+    sed -i '/^install-lockpath:/!b;n;c\       # pass' backend/Makefile.am
```

Thank you for this improvement!

Could you explain shortly what exactly the `!b;n;c\` does?

Also, will this fail silently if upstream changes, and if yes is there a way to make it fail loudly, like e.g. https://nixos.org/manual/nixpkgs/stable/#fun-substitute-replace-fail ?

Thanks!